### PR TITLE
[docs-only] flesh out migration docs

### PR DESCRIPTION
**Pull request by butonic**
Adding more details to the migration stages with sections on:
- Why?
- User impact
- steps
- verification
- rollback
- notes
- FAQ

I'll also add TODOs where I run into missing things.

### CODE REVIEW

File: docs/ocis/migration.md [hunk 1]
```diff
@@ -20,28 +20,131 @@ The migration happens in subsequent stages while the service is online. First al
...
+
+#### FAQ
+_Feel free to add you rquestion as a PR to this document using the link at the top of this page!_ 
+
+### Stage 3: _introduce oCIS interally_
+
+#### Why?
+Befor letting oCIS handle end user requests we will first make it available in the internal network. By subsequently adding services we can add functionality and verify the services work as intended.
+
+Start oCIS backend and make read only tests on existing data using the `owncloudsql` storage driver which will read (and write)
```
Comment (fschade) [inline docs/ocis/migration.md:89]: can we mention a guide on how to get `owncloudsql` configured

**Review by fschade — COMMENTED** [review]
good read, just a small mention

File: docs/ocis/migration.md [hunk 2]
```diff
@@ -10,38 +10,117 @@ geekdocFilePath: migration.md
...
+
+#### Notes
+The ownCloud 10 demo instance uses OAuth to obtain a token for ownCloud web. In oCIS the token is provided by the OpenID Connect Identityd Provider, which may skip the consent step for trusted clients for a seamless login experience.
+
+#### FAQ
+_Feel free to add your question as a PR to this document using the link at the top of this page!_ 
+
+### Stage 2: _introduce OpenID Connect_
+
+Basic auth requires us to properly store and manage user credentials. Something we would rather like to delegate to a tool specifically built for that task.
```
Comment (pmaier1) [inline docs/ocis/migration.md:46]: I wrote a blog post on this some time ago. Maybe you want to reference it https://owncloud.com/news/openid-connect-oidc-app/

**Comment by butonic** [comment]
@pmaier1 I added your link. @michl19 I updated the share manage migration stage with what came to my mind.

@dragotin @micbar the TODOs feels like I wrote another roadmap ... we should link them to tickets.

**Review by micbar — APPROVED** [review]

**Pull request merged**